src/storm/modelchecker/helper/DiscountingHelper.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include "SingleValueModelCheckerHelper.h"
+#include "storm/solver/helper/ValueIterationOperator.h"
+#include "storm/storage/Scheduler.h"
+#include "storm/utility/ProgressMeasurement.h"
+
+namespace storm {
+class Environment;
+namespace modelchecker {
+namespace helper {
+template<typename ValueType, bool TrivialRowGrouping = false>
+class DiscountingHelper : public SingleValueModelCheckerHelper<ValueType, storm::models::ModelRepresentation::Sparse> {
+   public:
+    DiscountingHelper(storm::storage::SparseMatrix<ValueType> const& A, ValueType discountFactor);
+    DiscountingHelper(storm::storage::SparseMatrix<ValueType> const& A, ValueType discountFactor, bool trackScheduler);
+
+    void setUpViOperator() const;
```

Comment:
Wouldn't it be simpler to make`setUpViOperator` private and call it within `solveWithDiscountedValueIteration`?